packages/diagrams/backend/sirius-components-collaborative-diagrams/src/main/java/org/eclipse/sirius/components/collaborative/diagrams/handlers/DeleteFromDiagramEventHandler.java

```diff
@@ -249,23 +234,32 @@ private IStatus invokeDeleteEdgeTool(Edge edge, IEditingContext editingContext,
         return result;
     }
 
+    private VariableManager populateVariableManager(IEditingContext editingContext, IDiagramContext diagramContext, Optional<Node> node, Optional<Edge> edge, Optional<Object> self, DeletionPolicy deletionPolicy) {
```

Comment:
I'll just remove the optional from the signature here, I'm not fan of optionals in method signature, it just leads to errors. Here `null` would not create any issue for the selectedNode / selectedEdge and self does not need to be an optional because this method cannot work with anything expect a non-empty optional.